ApollosProject/apollos-prototype
# Add nested navigation to Content Single screens

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds nested navigation for navigating from one content-single screen to another:

![screen recording 2018-10-29 at 04 59 pm](https://user-images.githubusercontent.com/103927/47683029-4becee80-db9c-11e8-9deb-b3dfc1b3e8a6.gif)

Why is it needed:

- Better navigation from within content singles, ability to go back one level or completely close the modal from any level of nesting

### What design trade-offs/decisions were made?

The `content-single/NavigationHeader` component is a little crufty...but only solution available that I could find with `react-navigation` and nested navigtors, womp womp.

### How do I test this PR?

Open content single views and navigate to nested navigators. Aside: do we navigate to anywhere else from within the modal view? Should we implement this at a different level (right now the back+close actions only work from `ContentSingle` to `ContentSingle` navigations).

### What automated tests did you write?

Added a snapshot test for `ModalView` and the new `NavigationHeader`

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
